src/main/java/com/google/gcs/sdrs/service/worker/rule/impl/StsRuleExecutor.java

```diff
@@ -182,122 +186,236 @@ private StsRuleExecutor() throws IOException {
     List<RetentionJob> datasetRuleJobs = new ArrayList<>();
     // get all dataset rules for a bucket
     Map<String, List<RetentionRule>> bucketDatasetMap = buildBucketRuleMap(datasetRules);
-    String correlationId = getCorrelationId();
```

Comment:
why remove the correlationId?

Reply:
It's removed here but getCorrelationId will be called when needed..